ci-automation/vendor_test.sh

```diff
@@ -0,0 +1,204 @@
+ciavts_tapfile="${PWD}/${ciavts_work_dir}/${ciavts_tapfile}"
+ciavts_torcx_manifest="${PWD}/${ciavts_main_work_dir}/torcx_manifest.json"
+
+echo "++++ Running ${ciavts_testscript} inside ${ciavts_work_dir} ++++"
+
+cd "${ciavts_work_dir}"
+
+CIA_VERNUM="${ciavts_vernum}"
+CIA_ARCH="${ciavts_arch}"
+CIA_TAPFILE="${ciavts_tapfile}"
+CIA_CHANNEL="${ciavts_channel}"
+CIA_TESTSCRIPT="${ciavts_testscript}"
+CIA_GIT_VERSION="${ciavts_git_version}"
+CIA_BUILD_TYPE="${ciavts_type}"
+CIA_TORCX_MANIFEST="${ciavts_torcx_manifest}"
+
+# Unset all variables with ciavts_ prefix now.
+unset -v "${!ciavts_@}"
+
+# Prefixes all test names in the tap file with a given prefix, so the
+# test name like "cl.basic" will become "extra-test.${prefix}.cl.basic".
+#
+# Parameters:
+# 1 - prefix, all dots in it will be removed.
+# 2 - tap file, modified in place.
+function prefix_tap_file() {
+    local prefix="${1}"; shift
+    local tap_file="${1}"; shift
+    # drop the dots from prefix
+    local actual_prefix="extra-test.${prefix//.}."
+
+    sed --in-place --expression 's/^\(\s*\(not\)\?\s*ok[^-]*\s*-\s*\)\(\S\)/\1'"${prefix}"'\3/g' "${tap_file}"
+}
+
+# Filters the test names, so it only prints the real names of the
+# prefixed tests. For example for prefix "foo", it will ignore the
+# test name like "cl.basic", but will print "cl.internet" for a test
+# name like "extra-test.foo.cl.internet". "*" is treated specially -
+# it will be printed if it is passed.
+#
+# Typical use:
+#   tests_to_run=( $(filter_prefixed_tests "${instance_type}" "${@}") )
+#
+# Parameters:
+# 1 - prefix, all dots in it will be removed.
+# @ - test names
+function filter_prefixed_tests() {
+    local prefix="${1}"; shift
+    # rest of the parameters are test names
+    local name
+    local stripped_name
+    for name; do
+        stripped_name="${name#extra-test.${prefix//.}.}"
+        if [[ "${stripped_name}" != "${name}" ]]; then
+            echo "${stripped_name}"
+            continue
+        elif [[ "${name}" = '*' ]]; then
+            echo '*'
+        fi
+    done
+}
+
+# Filters the extra tests from the passed test names. Ignored test
+# names begin with "extra-test.".
+#
+# Typical use:
+#   tests_to_run=( $(filter_out_prefixed_tests "${@}") )
+#
+# Parameters:
+# @ - test names
+function filter_out_prefixed_tests() {
+    local name
+    for name; do
+        if [[ "${name#extra-test.}" = "${name}" ]]; then
+            echo "${name}"
+        fi
+    done
+}
+
+# Merges into the first (main) tap file the contents of other tap
+# files. It is very simple, so the function assumes that all the tap
+# files have the following contents:
+#
+# 1..${number_of_tests}
+# (not)? ok - ${test_name}
+# …
+#
+# The other tap files should already be preprocessed by
+# prefix_tap_file to avoid duplicated test names.
+#
+# Typical use:
+#   merge_tap_files "${tap_file}" extra-tap-files-*
+#
+# Parameters:
+# 1 - main tap file
+# @ - other tap files
+function merge_tap_files() {
```

Comment:
Good idea to provide a function for this here.

I think we could try to move the whole `for instance_type in …` logic here, too. For simplicity we can keep the `r